src/protocol/context.rs

```diff
@@ -177,3 +177,74 @@ impl<'a, F: Field> ProtocolContext<'a, MaliciousReplicated<F>, F> {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{
+        error::BoxError,
+        ff::Fp31,
+        protocol::QueryId,
+        test_fixture::{make_contexts, make_world},
+    };
+
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+    struct TestStep;
+
+    impl Substep for TestStep {}
+
+    impl AsRef<str> for TestStep {
+        fn as_ref(&self) -> &str {
+            "test"
+        }
+    }
+
+    #[tokio::test]
+    pub async fn narrow() -> Result<(), BoxError> {
```

Comment:
it seems that you don't need those tests to be async?

Reply:
I tried changing them to regular tests, but there is a call to `tokio::spawn` within `InMemoryNetwork::new` within `make_world`.